plugins/modules/check.py

```diff
@@ -126,58 +130,71 @@
 '''
 
 EXAMPLES = '''
+- name: Create a check
+  check:
+    name: check
+    command: /bin/true
+    subscriptions:
+      - checks
+    interval: 30
 '''
 
 RETURN = '''
+object:
+    description: object representing Sensu check
+    returned: success
+    type: dict
 '''
 
 from ansible.module_utils.basic import AnsibleModule
-from ansible_collections.sensu.sensu_go.plugins.module_utils.base import SensuObject
+from ansible_collections.sensu.sensu_go.plugins.module_utils import (
+    arguments, errors, utils
+)
 
 
-class SensuCheck(SensuObject):
-    def __init__(self, module):
-        super(SensuCheck, self).__init__(module)
+def build_api_payload(params):
+    payload = arguments.get_mutation_payload(
+        params, 'command', 'subscriptions', 'handlers', 'interval', 'cron', 'publish', 'timeout',
+        'ttl', 'stdin', 'low_flap_threshold', 'high_flap_threshold', 'proxy_entity_name',
+        'output_metric_format', 'round_robin'
+    )
+    payload['check_hooks'] = utils.dict_to_single_item_dicts(params['check_hooks'])
+    payload['env_vars'] = utils.dict_to_key_value_strings(params['env_vars'])
 
-        self.path = '/checks/{0}'.format(self.params['name'])
+    # Special cases when even if it's empty, None is still returned from server
+    payload['runtime_assets'] = params['runtime_assets']
+    payload['output_metric_handlers'] = params['output_metric_handlers']
+    payload['subdue'] = None  # Not implemented by sensu, but still returned from API
 
-        for key in (
-            'command',
-            'subscriptions',
-            'handlers',
-            'interval',
-            'cron',
-            'publish',
-            'timeout',
-            'ttl',
-            'stdin',
-            'low_flap_threshold',
-            'high_flap_threshold',
-            'runtime_assets',
-            'proxy_entity_name',
-            'output_metric_format',
-            'output_metric_handlers',
-            'round_robin',
-        ):
-            if self.params[key] is not None:
-                self.payload[key] = self.params[key]
+    # Apparently empty string is also a valid choice and it's the default
+    payload['output_metric_format'] = payload.pop('output_metric_format', '')
 
-        self.param_dict_to_payload_list('check_hooks')
-        self.param_dict_to_payload_kv_list('env_vars')
+    if params['proxy_entity_attributes']:
+        payload['proxy_requests'] = {
+            'entity_attributes': params['proxy_entity_attributes'],
+            'splay': params['proxy_splay'],
```

Comment:
"splay" value is not a required parameter. In case user didn't set "proxy_splay" explicitly we shouldn't be sending that key at all. Sorry, I know it complicates code a little :)